quarkus/dist/pom.xml

```diff
@@ -60,7 +60,7 @@
     </dependencies>
 
     <build>
-        <finalName>keycloak-${project.version}</finalName>
+        <finalName>${dist.archive.file}</finalName>
```

Comment:
Worth noting that this only affects the name inside `target/`, the installed/deployed name still conforms to the maven repo specification. Although, it doesn't hurt either.

For example, with `-Ddist.archive.base.file.name=xyzzy`:

```
aszczucz@tyrfing ~/s/k/keycloak (zip_name_change)> ls -l quarkus/dist/target/*.zip
-rw-r--r--. 1 aszczucz aszczucz 166047756 Oct 25 08:22 quarkus/dist/target/xyzzy-999.0.0.zip
aszczucz@tyrfing ~/s/k/keycloak (zip_name_change)> ls -l ~/.m2/repository/org/keycloak/keycloak-quarkus-dist/999.0.0-SNAPSHOT/*.zip
-rw-r--r--. 1 aszczucz aszczucz 166047756 Oct 25 08:22 /home/aszczucz/.m2/repository/org/keycloak/keycloak-quarkus-dist/999.0.0-SNAPSHOT/keycloak-quarkus-dist-999.0.0-SNAPSHOT.zip
```

Reply:
I think this is OK to have different name in target/ . Processes using maven artifact download has to deal with it. The archive internal base dir will be correct.